### PR TITLE
Add test and docs for add external force function

### PR DESCRIPTION
**Pull request by CassianoMontibeller**

### CODE REVIEW

File: docs/Components/BaseComponent.md [hunk 1]
```diff
@@ -83,3 +83,18 @@ This property returns a dictionary with the hierarchical relationship of each co
 print(base_component.nested_components)
 >>> {'component': BaseComponent(name='component', type='generic_component', mass=3.4, children={'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={})}, external_forces={}, external_moments={}), 'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={})}
 ```
+## add_external_force_function
+This function is responsible for adding an external force applied to a given point on the component
```
Comment (rafaellehmkuhl) [inline docs/Components/BaseComponent.md:87]: This *method is responsible for *appending an external force *function to the component. *The force function returns a force vector and an application point.

File: tests/Components/test_BaseComponent.py [hunk 2]
```diff
@@ -99,6 +108,7 @@ def test_empty_mass(freebody_component, base_component, attached_component, payl
     assert(freebody_component.empty_mass == 4.8)
     assert(attached_component.empty_mass == 1.4)
 
+
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:111]: Remove this line.

File: tests/Components/test_BaseComponent.py [hunk 3]
```diff
@@ -53,6 +53,15 @@ def payload():
...
 
+@pytest.fixture
+def force1():
+    mag = 10
+    ang = math.radians(45)
+    force_point = Vector2(-10, 0)
+    force1 = Vector2(r=mag, theta=ang)
+    return force1, force_point
+
+
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:64]: Nice

File: tests/Components/test_BaseComponent.py [hunk 4]
```diff
@@ -178,3 +188,9 @@ def test_nested_components(base_component, attached_component):
            'component': base_component, 'attached_component': attached_component})
     assert(attached_component.nested_components == {
            'attached_component': attached_component})
+
+
+def test_add_external_force_function(force1, base_component):
+    base_component.add_external_force_function('force1', force1)
+    assert(base_component.external_forces['force1'] == force1)
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:195]: Perfect!

File: tests/Components/test_BaseComponent.py [hunk 5]
```diff
@@ -178,3 +188,9 @@ def test_nested_components(base_component, attached_component):
...
+    assert(base_component.external_forces == {'force1': force1})
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:196]: This assert is not needed

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

**Review by rafaellehmkuhl — APPROVED** [review]

**Review by joaofoes — APPROVED** [review]

**Pull request merged**